File: Graphs/AdjacencyMatrix.py

class Graph:
    def __init__(self,size):
        self.size = size 
        self.matrix = [[0] * size for _ in range(size)]
        self.vertex_data = [" "] * size 
    def addEgde(self,i,j):
        if 0 <= i < self.size and 0 <= j < self.size:
            self.matrix[i][j] = 1
            self.matrix[j][i] = 1
    def addVertex(self,vertex,data):
        if 0 <= vertex < self.size:
            self.vertex_data[vertex] = data
    def dfs_util(self,v , visited):
        visited[v] = True 
        print(self.vertex_data[v] , end = " ")

        for i in range(self.size):
            if self.matrix[v][i] == 1 and not visited[i]:
                self.dfs_util(i,visited)
    def dfs(self,startVertex):
        visited = [False] * self.size
        startVertex = self.vertex_data.index(startVertex)
        self.dfs_util(startVertex,visited)
    def bfs(self, startVertexData):
        queue = [self.vertex_data.index(startVertexData)]
        visited = [False] * self.size
        visited[queue[0]] = True

        while queue:
            current_vertex = queue.pop(0)
            print(self.vertex_data[current_vertex], end=' ')
            for i in range(self.size):
                if self.matrix[current_vertex][i] is not None and not visited[i]:  # Edge exists
                    queue.append(i)
                    visited[i] = True
    def display(self):
        print("Adjacency Matrix")
        for row in self.matrix:
                print(" ".join(map(str,row)))
        print("\nVertex Data")
        for vertex , data in enumerate(self.vertex_data):
            print(f"Vertex {vertex} : {data}")

File: Graphs/AdjacencyMatrix.java
package DSA;

import java.util.*;

public class Graphs {
	int size;
	int[][] matrix;
	String[] vertex_data;

	public Graphs(int size) {
		this.size = size;
		this.matrix = new int[size][size];
		this.vertex_data = new String[size];
	}

	public void addEdge(int i, int j) {
		if ((i >= 0 && i < size) && (j >= 0 && j < size)) {
			matrix[i][j] = 1;
			matrix[j][i] = 1;
		} else {
			System.out.println("Loi");
			System.exit(0);
		}
	}

	public void addVertex(int vertex, String data) {
		if (vertex >= 0 && vertex < size) {
			vertex_data[vertex] = data;
		} else {
			System.out.println("Loi");
			System.exit(0);
		}
	}

	public boolean isConnected(int i, int j) {
		if (i < 0 || i >= size || j < 0 || j >= size) {
			return false;
		}
		return matrix[i][j] == 1;
	}

	public void dfs_util(int v, boolean[] visited) {
		visited[v] = true;
		System.out.print(vertex_data[v] + " ");
		for (int i = 0; i < size; i++) {
			if (matrix[v][i] == 1 && !visited[i])
				dfs_util(i, visited);
		}
	}

	public void dfs(int startVertex) {
		boolean[] visited = new boolean[size];
		dfs_util(startVertex, visited);
	}

	public void bfs(int startVertex) {
		boolean[] visited = new boolean[size];
		Queue<Integer> queue = new LinkedList<>();
		queue.add(startVertex);
		visited[startVertex] = true;
		while (!queue.isEmpty()) {
			int currentVertex = queue.poll();
			System.out.print(vertex_data[currentVertex] + " ");
			for (int i = 0; i < size; i++) {
				if (matrix[currentVertex][i] != 0 && !visited[i])
					queue.add(i);
				visited[i] = true;
			}
		}
	}

	public void display() {
		System.out.println("Vertex Data :");
		for (int vertex = 0; vertex < size; vertex++) {
			System.out.printf("Vertex %d: %s%n", vertex, vertex_data[vertex]);
		}

		System.out.println("Adjacency Matrix :");
		for (int[] row : matrix) {
			for (int value : row) {
				System.out.print(value + " ");
			}
			System.out.println();
		}

	}
}
